vpr/src/timing/net_delay.cpp

```diff
@@ -118,399 +68,76 @@ vtr::vector<ClusterNetId, float*> alloc_net_delay(vtr::t_chunk* chunk_list_ptr)
-     * compute the Tdel to each node in the rc tree.  The T_arrival is the time *
-     * at which the signal hits the input to this node.  This routine calls     *
-     * itself recursively to perform the traversal.                             */
-
-    float Tdel, Rmetal, Tchild;
-    t_linked_rc_edge* linked_rc_edge;
-    t_rc_node* child_node;
-    short iswitch;
+    auto& cluster_ctx = g_vpr_ctx.clustering();
     int inode;
 
-    auto& device_ctx = g_vpr_ctx.device();
-
-    Tdel = T_arrival;
-    inode = rc_node->inode;
-    Rmetal = device_ctx.rr_nodes[inode].R();
-
-    /* NB:  device_ctx.rr_nodes[inode].C gives the capacitance of this node, while          *
-     * rc_node->C_downstream gives the unbuffered downstream capacitance rooted *
-     * at this node, including the C of the node itself.  I want to multiply    *
-     * the C of this node by 0.5 Rmetal, since it's a distributed RC line.      *
-     * Hence 0.5 Rmetal * Cnode is a pessimistic estimate of delay (i.e. end to *
-     * end).  For the downstream capacitance rooted at this node (not including *
-     * the capacitance of the node itself), I assume it is, on average,         *
-     * connected halfway along the line, so I also multiply by 0.5 Rmetal.  To  *
-     * be totally pessimistic I would multiply the downstream part of the       *
-     * capacitance by Rmetal.  Play with this equation if you like.             */
-
-    /* Rmetal is distributed so x0.5 */
-    Tdel += 0.5 * rc_node->C_downstream * Rmetal;
-    rc_node->Tdel = Tdel;
+    t_rt_node* rt_root = traceback_to_route_tree(net_id); // obtain the root of the tree constructed from the traceback
+    load_new_subtree_R_upstream(rt_root);                 // load in the resistance values for the route tree
+    load_new_subtree_C_downstream(rt_root);               // load in the capacitance values for the route tree
+    load_route_tree_Tdel(rt_root, 0.);                    // load the time delay values for the route tree
+    load_one_net_delay_recurr(rt_root, net_id);           // recursively traverse the tree and load entries into the inode_to_Tdel map
 
-    /* Now expand the children of this node to load their Tdel values.       */
-
-    linked_rc_edge = rc_node->u.child_list;
-
-    while (linked_rc_edge != nullptr) { /* For all children */
-        iswitch = linked_rc_edge->iswitch;
-        child_node = linked_rc_edge->child;
-
-        Tchild = Tdel + device_ctx.rr_switch_inf[iswitch].R * child_node->C_downstream;
-        Tchild += device_ctx.rr_switch_inf[iswitch].Tdel; /* Intrinsic switch delay. */
-        load_rc_tree_T(child_node, Tchild);
-
-        linked_rc_edge = linked_rc_edge->next;
+    for (unsigned int ipin = 1; ipin < cluster_ctx.clb_nlist.net_pins(net_id).size(); ipin++) {
+        inode = route_ctx.net_rr_terminals[net_id][ipin];                // look for the inode corresponding to the net_delay indices
```

Comment:
Comment should say rr node index (not inode) that corresponds to the sink that was used to route a certain connection. inode comment not much more descriptive than the variable name.